### DISCUSSION TITLE
A way to clone/edit the built in search engine ?

### DISCUSSION THREAD

**Discussion by NattanCochet** [squidfunk/mkdocs-material#5829] [Q&A] [answered]
Hello, this is my first time asking for help on Github Discussions.

Im currently working on updating a company's intern documentation and turned to mkdocs-material due to its easy to use interface and good looking visuals.
The company is asking me to make it so that the search engine can be editable if they want to add new stuff and I would like to know if its possible to take the built-in search engine and make it its own "New Plugin" so that I can edit it easily.

So my question is the following : Is there a way to clone the built-in search engine into its own plugin or make it so that I can simply add code (not just the options in the yml files) to the existing engine and update it to whatever they would like ?

Thanks in advance!

**Comment by squidfunk** [comment, marked as answer]
Absolutely possible! You could extract the search plugin into a separate project and manage from there. Note that search consists of two parts: index generation (search plugin) and the actual search UI (frontend). The search UI is part of Material for MkDocs and likely harder to cut out cleanly, but it should be possible as well. Another idea might be to just fork the entire repository and change the parts you need adjusted ☺️

**Reply by NattanCochet** [reply]
Thank you for your answer!
I'll go check it out! I'll close the discussion if I find the things I need

Regards
